python/ray/data/datasource/parquet_datasource.py

```diff
@@ -459,7 +463,11 @@ def _estimate_files_encoding_ratio(self) -> float:
             # Use SPREAD scheduling strategy to avoid packing many sampling tasks on
             # same machine to cause OOM issue, as sampling can be memory-intensive.
             futures.append(
-                sample_fragment.options(scheduling_strategy=scheduling).remote(
+                sample_fragment.options(
+                    scheduling_strategy=scheduling,
+                    # Retry in case of transient errors during sampling.
+                    retry_exceptions=[OSError],
```

Comment:
Let's extract this as a variable similar to `RETRY_EXCEPTIONS_FOR_META_FETCH_TASK`, so we can tune later.

Reply:
discussed with @scottjlee , we'll make this configurable in DataContext. Since other retry errors are not configurable either. we'll use a follow-up PR to update all.

Reply:
I feel it's okay to not put into DataContext. I don't think they are user-facing, and mostly it only should be tuned by ourselves, when debugging user job. Let's only add config to DataContext if they are expected to be set by users.

Reply:
But agreed to merge this PR ASAP to unblock users.